cmd/state-exec/meta.go

```diff
@@ -37,24 +37,21 @@ func newExecutorMeta(execPath string) (*executorMeta, error) {
 		return nil, fmt.Errorf("cannot get matching bin by path: %w", err)
 	}
 
-	transformEnv := true
+	onlyTransformPath := false
 	if projectDir := os.Getenv(constants.ActivatedStateEnvVarName); projectDir != "" {
 		// Note: cannot import runtime/target for ProjectDirToTargetDir() because that would bloat
 		// up the exe from <4MB to >15MB.
 		projectCacheDir := filepath.Join(storage.CachePath(), hash.ShortHash(projectDir))
-		if projectCacheDir == execDir {
-			transformEnv = false // runtime environment variables already set
+		if projectCacheDir == filepath.Dir(execDir) {
```

Comment:
I realize this wasn't introduced in this PR, but it just occurred to me that this isn't going to work if someone checked out with a custom path.

Additionally, we can't assume the two sides of the path were constructed the same way. We get around this in other places with this function:

https://github.com/ActiveState/cli/blob/d6bc46198b35e47cbe7958e5b935482f09d17008/internal/fileutils/fileutils.go#L1165-L1165